packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SubModelTree/index.tsx

```diff
@@ -55,11 +56,16 @@ const SubModelTree: FC<SubModelTreeProps> = ({
 
   const onCreate = useCallback(() => {
     // Prevent duplicates
-    const duplicates = Object.entries(document.nodeMap).filter(
-      ([name, node]) => node.properties?.subModelId === object3D.name,
-    );
-
-    if (duplicates.length > 0) {
+    const duplicates = getSceneNodeByRef(parentRef)?.childRefs.filter((childRef) => {
+      const child = getSceneNodeByRef(childRef);
+      const childSubModelComponent = findComponentByType(
+        child,
+        KnownComponentType.SubModelRef,
+      ) as ISubModelRefComponent;
+      return childSubModelComponent?.selector === object3D.name;
+    });
+
+    if (duplicates && duplicates.length > 0) {
```

Comment:
You can implement optional chaining here `if (duplicates?.length > 0)`.

Reply:
I did that but it fails because undefined cannot be checked for > 0 apparently

Reply:
Or it is an unreliable check.

Reply:
True. 1undefined` is true in javascript. If that's what you're working around, then it should be updated to `duplicates !== undefined && duplicates.length > 0` for readability & preventing linters from re-writing it.